pkg/logs/internal/launchers/container/tailerfactory/file.go

```diff
@@ -152,8 +159,16 @@ func (tf *factory) findDockerLogPath(containerID string) string {
 				podmanLogsBasePath, "storage/overlay-containers", containerID,
 				"userdata/ctr.log")
 		}
+		// if the user has set a custom docker data root, this will pick it up
+		// and set it in place of the usual docker base path
+		dockerLogsPath := dockerLogsBasePathNix
```

Comment:
Instead of duplicating this logic twice for windows and non-windows. Since the path join logic is the same, perhaps you could move this out above the `switch`? 

Something like 

```go
overridePath := coreConfig.Datadog.GetString("logs_config.docker_path_override")
if len(overridePath) > 0 {
    return filepath.Join(...
}

switch runtime.GOOS {
...
```